legion/sdk/legion/sdk/clients/edi.py

```diff
@@ -182,102 +220,45 @@ def _request(self,
             else:
                 LOGGER.debug('Got response. Breaking')
                 break
-
             left_retries -= 1
         else:
             raise EDIConnectionException('Can not reach {}'.format(self._base_url)) from raised_exception
 
-        # We assume if there were redirects then credentials are out of date and we can refresh or build auth url
-        if response.history:
-            # If it is a error after refreshed token - fail
-            if limit_stack:
-                raise IncorrectAuthorizationToken(
-                    f'{credentials_error_status} even after refreshing. \n'
-                    'Please try to log in again'
-                ) from raised_exception
-
-            LOGGER.debug('Status code: "{}", Response: "{}"'.format(response.status_code, response.text))
-            LOGGER.debug('Redirect has been detected. Trying to refresh a token')
-
-            if legion.sdk.config.EDI_REFRESH_TOKEN and legion.sdk.config.EDI_ISSUING_URL:
-                LOGGER.debug('Refresh token for %s has been found, trying to use it', legion.sdk.config.EDI_ISSUING_URL)
-                login_result = do_refresh_token(legion.sdk.config.EDI_REFRESH_TOKEN, legion.sdk.config.EDI_ISSUING_URL)
-                if not login_result:
-                    raise IncorrectAuthorizationToken(
-                        'Refresh token in not correct. \n'
-                        'Please login again'
-                    ) from raised_exception
-                else:
-                    self._update_config_with_new_oauth_config(login_result)
-                    return self._request(
-                        url_template,
-                        payload=payload,
-                        action=action,
-                        stream=stream,
-                        timeout=timeout,
-                        limit_stack=True
-                    )
-            else:
-                if self._non_interactive:
-                    raise IncorrectAuthorizationToken(
-                        f'{credentials_error_status}. \n'
-                        'Please provide correct temporary token or disable non interactive mode'
-                    ) from raised_exception
-                else:
-                    # Start interactive flow
-                    self._interactive_login_finished.clear()
-                    target_url = get_authorization_redirect(response.url, self.after_login)
-                    LOGGER.error('%s. \nPlease open %s', credentials_error_status, target_url)
-
-                    self._interactive_login_finished.wait()
-                    return self._request(
-                        url_template,
-                        payload=payload,
-                        action=action,
-                        stream=stream,
-                        timeout=timeout,
-                        limit_stack=True
-                    )
-
-        return response
+        if self._login_required(response):
```

Comment:
All our security functions, which interacts with a Keycloak, are synchronous. Is it expected?
`legion/jyputerlab/handlers/oauth2.py` is too.